wandb/cli/cli.py

```diff
@@ -1445,7 +1455,34 @@ def launch_agent(
     max_jobs=None,
     config=None,
     url=None,
+    log_file=None,
 ):
+    if log_file is not None:
+        _launch_logger = logging.getLogger("wandb.sdk.launch")
+        if log_file == "-":
+            log_file_stream = sys.stdout
+        else:
+            try:
+                log_file_stream = open(log_file, "w")
+            # check if file is writable
+            except Exception as e:
+                wandb.termerror(
+                    f"Could not open {log_file} for writing logs. Please check "
+                    f"the path and permissions.\nError: {e}"
+                )
+                return
+
+        wandb.termlog(
+            f"Logging agent output to {'stdout' if log_file == '-' else log_file}. "
+        )
+        handler = logging.StreamHandler(log_file_stream)
+        handler.formatter = logging.Formatter(
+            "%(asctime)s %(levelname)-7s %(threadName)-10s:%(process)d "
+            "[%(filename)s:%(funcName)s():%(lineno)s] %(message)s"
+        )
+        _launch_logger.addHandler(handler)
+        del _launch_logger
```

Comment:
TIOLI: I consider this kind of manual scope management a code smell that this block should be extracted to a separate function

Reply:
I'm happy to extract- does that make this not smelly? Or does the scope management still stink